pkg/controllers/provisioning/v1alpha1/allocation/constraints.go

```diff
@@ -41,7 +40,7 @@ func (c *Constraints) Group(ctx context.Context, provisioner *v1alpha1.Provision
 	// Groups uniqueness is tracked by hash(NodeConstraints)
 	groups := map[uint64]*cloudprovider.Constraints{}
 	for _, pod := range pods {
-		constraints := c.getConstraints(provisioner, pod)
+		constraints := provisioner.ConstraintsWithOverrides(pod)
 		key, err := hashstructure.Hash(constraints, hashstructure.FormatV2, nil)
```

Comment:
I'm wondering if this isn't quite enough - eg if lists in the sub-struct contain same elements but are sorted differently for some reason?

Reply:
Let me play with this